core-plugins/src/test/java/co/cask/hydrator/plugin/batch/action/HDFSActionTestRun.java

```diff
@@ -0,0 +1,144 @@
+import co.cask.cdap.etl.proto.v2.ETLPlugin;
+import co.cask.cdap.etl.proto.v2.ETLStage;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramRunStatus;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.cdap.proto.RunRecord;
+import co.cask.cdap.proto.WorkflowNodeStateDetail;
+import co.cask.cdap.proto.artifact.AppRequest;
+import co.cask.cdap.test.ApplicationManager;
+import co.cask.cdap.test.WorkflowManager;
+import co.cask.hydrator.plugin.batch.ETLBatchTestBase;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Test for {@link HDFSAction}
+ */
+public class HDFSActionTestRun extends ETLBatchTestBase {
+  @ClassRule
+  public static TemporaryFolder folder = new TemporaryFolder();
+
+  @Test
+  public void testFailedHDFSAction() throws Exception {
+
+    folder.newFolder("source");
+    folder.newFile("source/test.txt");
+    folder.newFile("source/test.json");
+
+    ETLStage action = new ETLStage(
+      "HDFSAction",
+      new ETLPlugin("HDFSAction", Action.PLUGIN_TYPE,
+                    ImmutableMap.of("sourcePath", folder.getRoot() + "/source/random",
+                                    "destPath", folder.getRoot() + "/",
+                                    "fileRegex", ".*\\.txt",
+                                    "continueOnError", "false"),
+                    null));
+    ETLStage source = new ETLStage("source",
+                                   new ETLPlugin("KVTable", BatchSource.PLUGIN_TYPE,
+                                                 ImmutableMap.of("name", "hdfsTestSource"), null));
+    ETLStage sink = new ETLStage("sink", new ETLPlugin("KVTable", BatchSink.PLUGIN_TYPE,
+                                                       ImmutableMap.of("name", "hdfsTestSink"), null));
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(sink)
+      .addStage(action)
+      .addConnection(action.getName(), source.getName())
+      .addConnection(source.getName(), sink.getName())
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(DATAPIPELINE_ARTIFACT, etlConfig);
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, "hdfsActionTest");
+    ApplicationManager appManager = deployApplication(appId, appRequest);
+    WorkflowManager manager = appManager.getWorkflowManager(SmartWorkflow.NAME);
+    manager.start(ImmutableMap.of("logical.start.time", "0"));
+    manager.waitForFinish(3, TimeUnit.MINUTES);
+
+    List<RunRecord> history = appManager.getHistory(new Id.Program(appId, ProgramType.WORKFLOW, SmartWorkflow.NAME),
+                                                    ProgramRunStatus.FAILED);
+    Assert.assertTrue(history.size() == 1); //make sure pipeline didn't fail
```

Comment:
o yeah you're right, didn't see the ProgramRunStatus.FAILED. I guess its just a misleading comment then.